Assets/Mirror/Runtime/NetworkWriter.cs

```diff
@@ -102,16 +102,16 @@ public void Write(byte[] buffer, int offset, int count)
         // (like an inventory with different items etc.)
         public void WriteBytesAndSize(byte[] buffer, int offset, int count)
         {
-            uint length = checked((uint)count);
             // null is supported because [SyncVar]s might be structs with null byte[] arrays
             // (writing a size=0 empty array is not the same, the server and client would be out of sync)
             // (using size=-1 for null would limit max size to 32kb instead of 64kb)
-            writer.Write(buffer != null); // notNull?
-            if (buffer != null)
+            if (buffer == null)
             {
-                WritePackedUInt32(length);
-                writer.Write(buffer, offset, count);
+                WritePackedUInt32(0u);// notNull?
```

Comment:
Per Microsoft's guidelins,  comments go before the line,  not at the end of it

Reply:
Was like that earlier, didn't want to change it

Reply:
```suggestion
                WritePackedUInt32(0u);
```